test/dma_tests.py

```diff
@@ -542,6 +543,121 @@ def gain_check(uri, classname, channel, param_set, dds_scale, min_rssi, max_rssi
     assert rssi >= min_rssi
     assert rssi <= max_rssi
 
+def hardwaregain(uri, classname, channel, dds_scale, frequency, hardwaregain_low, hardwaregain_high):
+    """ test_hadwaregain: Test loopback with connected cables anf verify 
+        calculated hardwaregain. This is only applicable for devices with RSSI calculations
+        onboard. This test also requires a devices with TX and RX onboard where the
+        transmit signal can be recovered. TX FPGA DDSs are used to generate a
+        sinusoid which is then received on the RX side.
+
+        parameters:
+            uri: type=string
+                URI of IIO context of target board/system
+            classname: type=string
+                Name of pyadi interface class which contain attribute
+            channel: type=list
+                List of integers or list of list of integers of channels to
+                enable through tx_enabled_channels
+            dds_scale: type=float
+                Scale of DDS tone. Range [0,1]
+            frequency:
+                Frequency in hertz of the generated tone. This must be
+                less than 1/2 the sample rate.
+            hardwaregain_low: type=float
+                Minimum acceptable value of hardwaregain attribute
+            hardwaregain_high: type=float
+                Maximum acceptable value of hardwaregain attribute
+
+    """
+    sdr = eval(classname + "(uri='" + uri + "')")
+    sdr.dds_single_tone(frequency, dds_scale, channel)
+    time.sleep(3)
+    
+    chname = "voltage" + str(channel)
+    hwgain = sdr._get_iio_attr(chname,"hardwaregain", False, sdr._ctrl)
+    print(hwgain)
+    assert hardwaregain_low <= hwgain <= hardwaregain_high
+
+def harmonic_vals(classname, uri, channel, param_set, low, high, plot=False):
+    """ harmonic_vals: Test first five harmonics and check to be within
```

Comment:
The description needs an update. DMA is used not the DDSs

Reply:
I apologize, I updated it now